ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -57,6 +58,9 @@
   private final Time time;
   private final UUID sessionId = UUID.randomUUID();
 
+  private ReplicaId replicaId;
```

Comment:
can we make this and `ClusterManagerWriteStatusDelegate`  `final` and set them to `null` in the constructors that don't provide them?

Reply:
I was trying to reuse the original ctor in the new ctor which prohibited me from making them final, but maybe I can just create an init(..) private method that they both use and then I can make them final... I'll see if that works

Reply:
nope, cause the final methods need to be set within the ctor themselves instead of a method that they call... I could do what you're asking but it be dup'ing code